packages/gatsby/cache-dir/loader.js

```diff
@@ -290,8 +291,18 @@ export class BaseLoader {
           }
           // undefined if final result is an error
           return pageResources
-        }
-      )
+        })
+
+      let componentChunkPromise
+      if (
+        process.env.NODE_ENV === `development` &&
+        process.env.GATSBY_EXPERIMENTAL_LAZY_DEVJS &&
+        options.isPrefetch === true
+      ) {
+        componentChunkPromise = () => Promise.resolve()
+      } else {
+        componentChunkPromise = createLoadComponentPromise()
```

Comment:
I could be wrong but I believe in the first case the `componentChunkPromise` becomes an arrow that returns a promise and in the `else` case it becomes an actual promise?

```suggestion
        componentChunkPromise = Promise.resolve()
      } else {
        componentChunkPromise = createLoadComponentPromise()
```

Any chance any of this could be tested?